packages/dialog/theme/material/vaadin-dialog-styles.js

```diff
@@ -15,6 +15,19 @@ const dialogOverlay = css`
     padding: 24px;
   }
 
+  :host([has-header]) [part='header'],
+  :host([has-title]) [part='header'] {
+    padding: 9px 24px;
+    font-weight: 500;
+    gap: 8px;
+    border-bottom: 1px solid rgba(0, 0, 0, 0.12);
```

Comment:
Let's use divider color (please also add corresponding theme file import):

```suggestion
    border-bottom: 1px solid var(--material-divider-color);
```